tests/server/test_resourceservice.py

```diff
@@ -617,6 +617,7 @@ async def test_log_deploy_start(server, client, environment, clienthelper, agent
         {"name": "file1", "id": rvid_r1_v1, "requires": [], "purged": False, "send_event": False},
     ]
     await clienthelper.put_version_simple(resources, version)
+    await data.ResourcePersistentState.populate_for_version(environment=uuid.UUID(environment), model_version=version)
```

Comment:
Can you elaborate on these changes? It seems sensible to me, but I don't immediately see why it's suddenly required while it wasn't before. Is this because we now write to the RPS table when we start a deploy?

Reply:
Yes, we now write to the RPS table when we start a deploy. I don't know if I should be here or right before we write the state to the rps table

Reply:
Here looks fine to me